src/_P039_Thermosensors.ino

```diff
@@ -405,9 +423,21 @@ boolean Plugin_039(byte function, struct EventStruct *event, String& string)
           }
           {
             addFormNote(F("Set Thermocouple type for MAX31856"));
-            const String Toptions[8]      = { F("B"), F("E"), F("J"), F("K"), F("N"), F("R"), F("S"), F("T") };
-            const int    ToptionValues[8] = { 0, 1, 2, 3, 4, 5, 6, 7 };
-            addFormSelector(F("Thermocouple type"), F("P039_tctype"), 8, Toptions, ToptionValues, P039_TC_TYPE);
+            const String Toptions[10]      = { F("B"), F("E"), F("J"), F("K"), F("N"), F("R"), F("S"), F("T"), F("VM16"), F("VM32") };
+            const int    ToptionValues[10] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 12};
+            addFormSelector(F("Thermocouple type"), F("P039_tctype"), 10, Toptions, ToptionValues, P039_TC_TYPE);
+          }
+          {
+            const String Coptions[5] = {F("1 sample"), F("2 samples"), F("4 samples"), F("8 samples"), F("16 samples")};
+            const int CoptionValues[5] = {0, 1, 2, 3, 4};
+            addFormSelector(F("Averaging"), F("P039_contype"), 5, Coptions, CoptionValues, P039_CONFIG_4);
+            addFormNote(F("Set Averaging Type for MAX31856"));
+          }
+          {
+            const String FToptions[2] = {F("60 Hz"), F("50 Hz")};
+            const int FToptionValues[2] = {0, 1};
+            addFormSelector(F("Supply Frequency Filter"), F("P039_filttype"), 2, FToptions, FToptionValues, P039_RTD_FILT_TYPE);
+            addFormNote(F("Set filter frequency for supply voltage. Choose appropriate to your power net frequency (50/60 Hz)"));
```

Comment:
Ah the filter frequency selector is already present... great minds :)
Have to double check then to make sure it is actually being set.....

Reply:
Unsure what action to derive from this finding... i tested that the bit in the register will be changed if you change the configuration here...

Reply:
For me this is more a comment and i consider it OK.